### PR TITLE
feat: add multiarch build and release

### PR DESCRIPTION
**Pull request by kalidax**
This job will

- Build and push an image to dockerhub.

- Build tofnd binary, gpg sign it, zip it, create sha256 hashes for both zip and binary, upload files to release and S3.

- For binary build the job will run on each combination of OS/arch defined in the matrix.

- For the container build it will run one time only on ubuntu-18.04/amd64

- Multiarch binary tests without docker push, upload to release and upload to s3 have been done here : https://github.com/axelarnetwork/tofnd/actions/runs/1921406385

### CODE REVIEW

**Review by ggutoski — APPROVED** [review]
Approving as codeowner but @talalashraf or @erain9 should approve before merging.

This PR subsumes #274 so we can close that after merge.

File: .github/workflows/build-docker-image-and-binaries.yaml [hunk 1]
```diff
@@ -36,59 +37,102 @@ jobs:
           submodules: recursive
 
       - name: Install Cosign
+        if: matrix.os == 'ubuntu-18.04' && matrix.arch == 'amd64'
```
Comment (milapsheth) [inline .github/workflows/build-docker-image-and-binaries.yaml:40]: Why not sign all binaries?

File: .github/workflows/build-docker-image-and-binaries.yaml [hunk 2]
```diff
@@ -36,59 +37,102 @@ jobs:
...
         run: cosign sign --oidc-issuer https://token.actions.githubusercontent.com ${TAGS}
         env:
           TAGS: axelarnet/tofnd:${{ github.event.inputs.tag }}
           COSIGN_EXPERIMENTAL: 1
 
+      - name: Install Rust
+        run: |
+          curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+
+      - name: build binaries for Linux/MacOS
```
Comment (milapsheth) [inline .github/workflows/build-docker-image-and-binaries.yaml:78]:
```suggestion
      - name: Build binaries
```
Perhaps?

**Review by milapsheth — APPROVED** [review]

File: .github/workflows/build-docker-image-and-binaries.yaml (hunk 1 above)
Comment (kalidax) [reply .github/workflows/build-docker-image-and-binaries.yaml:40]: because it is related to docker images. Binaries are signed in gpg section

**Comment by kalidax** [comment]
These update will improve the build time.
- Added parallel builds : 
- 2 jobs will run in parallel : release-binaries (macos/amd64 and ubuntu/amd64) and release-docker
- I also added the check of an existing release in S3 for avoiding the overwrite

**Comment by talalashraf** [comment]
@kalidax @milapsheth @ggutoski can we get this across the line ?

**Pull request merged**